README.md
Waiting for Moranis
===================

<a href='https://play.google.com/store/apps/details?id=org.adrienbricchi.waitingformoranis'>
  <img alt='Get it on Google Play'
       src='https://play.google.com/intl/en_us/badges/static/images/badges/en_badge_web_generic.png'
       width='200'/>
</a>

## Summary

A simple app to retrieve movie releases dates.  
With a single list to keep track of the countdown.

* Add movies to your waiting list
* Refresh the list, retrieve release dates
* Setup a Google Calendar to synchronize movie releases as daily events
* Share your Google Calendar, keep it to yourself, or do whatever the Calendar app can do


## Privacy policy

We don't store your data, period.  
We don't want to, nor have the infrastructure for that.

If you use the Google Calendar (not required, and disabled by default), the Google privacy policy will apply.  
If you use your own TMDB account's API Key (not required, and disabled by default), the TMDB privacy policy will apply.


CHANGELOG.md
Changelog
=========

All notable changes to this project will be documented in this file.

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),  
and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).


## [1.3.5](https://github.com/adrienbricchi/waiting-for-moranis/releases/tag/1.3.5) - 2023-07-03
### Added
- Canceled movies parsing

## [1.3.4](https://github.com/adrienbricchi/waiting-for-moranis/releases/tag/1.3.4) - 2023-04-11
### Fixed
- Movies release date parse from TMDB


## [1.3.3](https://github.com/adrienbricchi/waiting-for-moranis/releases/tag/1.3.3) - 2021-10-21
### Fixed
- Movies date locale
- TV Shows pilot release date


## [1.3.2](https://github.com/adrienbricchi/waiting-for-moranis/releases/tag/1.3.2) - 2021-10-09
### Fixed
- TV shows release dates


## [1.3.1](https://github.com/adrienbricchi/waiting-for-moranis/releases/tag/1.3.1) - 2021-09-28
### Add
- Data auto upgrade


## [1.3.0](https://github.com/adrienbricchi/waiting-for-moranis/releases/tag/1.3.0) - 2021-08-29
### Fixed
- Android 11+ compatibility


## [1.2.0](https://github.com/adrienbricchi/waiting-for-moranis/releases/tag/1.2.0) - 2021-04-04
### Added
- Shows


## [1.1.6](https://github.com/adrienbricchi/waiting-for-moranis/releases/tag/1.1.6) - 2020-12-29
### Fixed
- Crash fixes


## [1.1.5](https://github.com/adrienbricchi/waiting-for-moranis/releases/tag/1.1.5) - 2020-09-25
### Fixed
- Crash fixes


## [1.1.4](https://github.com/adrienbricchi/waiting-for-moranis/releases/tag/1.1.4) - 2020-09-25
### Fixed
- Crash fixes


## [1.1.3](https://github.com/adrienbricchi/waiting-for-moranis/releases/tag/1.1.3) - 2020-09-22
### Fixed
- Crash fixes


## [1.1.2](https://github.com/adrienbricchi/waiting-for-moranis/releases/tag/1.1.2) - 2020-09-13
### Added
- More unit tests
### Fixed
- Crash fixes
- Lint warnings


## [1.1.1](https://github.com/adrienbricchi/waiting-for-moranis/releases/tag/1.1.1) - 2020-08-06
### Fixed
- Crash fixes


## [1.1.0](https://github.com/adrienbricchi/waiting-for-moranis/releases/tag/1.1.0) - 2020-07-07
### Added
- Edit movie release dates, on single selection
- Slow timeout on same-movie request (30min)
### Fixed
- Crash on same-release date movies
- Crash on patch 100 to 101


## [1.0.1](https://github.com/adrienbricchi/waiting-for-moranis/releases/tag/1.0.1) - 2020-06-04
### Changed
- Minor UI fixes


## [1.0.0](https://github.com/adrienbricchi/waiting-for-moranis/releases/tag/1.0.0) - 2020-05-25
### Added
- Main countdown list
- Add/remove movie
- Retrieve release dates from TMDB
- Add events to Google Calendar


app/src/main/java/org/adrienbricchi/waitingformoranis/MovieListFragment.java
/*
 * Waiting For Moranis
 * Copyright (C) 2020
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program. If not, see <http://www.gnu.org/licenses/>.
 */

package org.adrienbricchi.waitingformoranis;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ListView;
import android.widget.SimpleAdapter;
import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.Map;


public class MovieListFragment extends Fragment {

    private ListView movieListView;


    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        return inflater.inflate(R.layout.movie_list, container, false);
    }


    public void onViewCreated(@NonNull View view, Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        movieListView = view.findViewById(R.id.list_view);
        ArrayList<Map<String, String>> arrayList = new ArrayList<>();

        HashMap<String, String> hashMap = new HashMap<>();
        hashMap.put("name", "James Bond");
        hashMap.put("date", "02/12/20");
        arrayList.add(hashMap);

        HashMap<String, String> hashMap2 = new HashMap<>();
        hashMap2.put("name", "Avengers");
        hashMap2.put("date", "04/05/21");
        arrayList.add(hashMap2);

        HashMap<String, String> hashMap3 = new HashMap<>();
        hashMap3.put("name", "Star Wars");
        hashMap3.put("date", "09/08/21");
        arrayList.add(hashMap3);


        String[] from = {"name", "date"};
        int[] to = {R.id.movieTitleTextView, R.id.movieDateTextView};

        SimpleAdapter simpleAdapter = new SimpleAdapter(
                getContext(),
                arrayList,
                R.layout.movie_list_cell,
                from,
                to
        );

        movieListView.setAdapter(simpleAdapter);
    }

}
